pandas/core/generic.py

```diff
@@ -4260,16 +4271,76 @@ def _get_values(self):
         return self.values
 
     def get_values(self):
-        """same as values (but handles sparseness conversions)"""
+        """
+        Same as values (but handles sparseness conversions).
+
+        Returns
+        -------
+        numpy.ndaray
+            Numpy representation of NDFrame
+
+        Examples
+        --------
+        >>> df = pd.DataFrame({'a': np.random.randn(2).astype('f4'),
+        ...                    'b': [True, False], 'c': [1.0, 2.0]})
+        >>> df.get_values()
+        array([[0.25209328532218933, True, 1.0],
+        [0.35383567214012146, False, 2.0]], dtype=object)
+        """
         return self.values
 
     def get_dtype_counts(self):
-        """Return the counts of dtypes in this object."""
+        """
+        Return counts of unique dtypes in this object.
+
+        Returns
+        -------
+        dtype   Number of dtype
+
+        See Also
+        --------
+        dtypes : Return the dtypes in this object.
+
+        Examples
+        --------
+        >>> a = [['a', 1, 1.0], ['b', 2, 2.0], ['c', 3, 3.0]]
+        >>> df = pd.DataFrame(a, columns=['str', 'int', 'float'])
+        >>> df['int'].astype(int)
```

Comment:
If you make `df` as from a dictionary the types should be correctly inferred and you won't need astypes